index.js

```diff
@@ -15,6 +34,37 @@ boredAppServer.get("/random", function (request, response) {
   response.send(randomActivity);
 });
 
+// Allow users to suggest new activities via http POST requests on /suggest
+boredAppServer.post("/suggest", async function (request, response) {
+
+  // The suggested activity is stored in the body of the http request
+  const suggestedActivity = request.body;
+
+  // Calling the function to check that the object passed to the funtion is valid
+  const isValid = isActivityValid(suggestedActivity);
+
+  //Validate the input from the user and return bad request if the input is invalid
+  if (isValid === false) {
+    response.status(400).send("Your suggestion is invalid. Please try again");
+    return;
+  }
+ 
+  // Add new suggested activity to an array
+  arrayOfSuggestedIdeas.push(suggestedActivity);
+
+  console.log(arrayOfSuggestedIdeas);
+
+  //Save the array of suggested activities to the suggestions file (suggestions.json)
+  const data = JSON.stringify(arrayOfSuggestedIdeas, null, 2);
+  const promise = writeFile("suggestions.json", data);
+  await promise;
```

Comment:
you can write this in one line like this
```suggestion
  await writeFile("suggestions.json", data);
```